transmart-notifications/grails-app/controllers/org/transmartproject/notifications/SubscriptionMailController.groovy

```diff
@@ -0,0 +1,36 @@
+package org.transmartproject.notifications
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.RequestParam
+import org.transmartproject.core.exceptions.InvalidArgumentsException
+import org.transmartproject.core.userquery.SubscriptionFrequency
+
+class SubscriptionMailController {
+
+    @Autowired
+    QuerySetSubscriptionMailService querySetSubscriptionMailService
+
+    /**
+     * GET /v2/admin/subscription/notify?frequency=DAILY|WEEKLY
+     *
+     *
+     * Sends emails to users who subscribed for a user query updates.
+     *
+     * @param frequency specifies whether the mail should be send to users
+     *        who subscribed for DAILY or WEEKLY subscription.
+     * @return {@link HttpStatus#OK};
+     *      or {@link HttpStatus#FORBIDDEN} when the user does not have an ADMIN role,
+     *      see request matcher configuration of the current application.
+     * @throws {@link InvalidArgumentsException} if frequency parameter is not valid.
+     */
+    def subscriptionNotify(@RequestParam('frequency')String frequency) {
+        SubscriptionFrequency subscriptionFrequency = frequency?.trim() ? SubscriptionFrequency.forName(frequency) : null
```

Comment:
Software development is a bit of a language game... looks to me that subscribing/subscription are one thing and sending notifications about updates another. I would prefer SubscriptionFrequency to be renamed to NotificationFrequency, etc to better reflect that distinction.

Reply:
The full renaming of SubscriptionFrequency would require changes in GB, should we do this before the release? I added some renaming to this controller and the added endpoint.